CONTRIBUTING.md

```diff
@@ -39,8 +39,46 @@ Or to be safe, you call apply all the required resources at once:
 
 1. `k-apply-all`
 
+### Releasing a New Version
+
+1. [Create Release Branch](#creating-a-release-branch) or [Patch a Release Branch](#patching-a-release-branch) the [Release Branch](#release-branches).
+1.
+Visit [Tag Workflow Action](https://github.com/crossplane-contrib/provider-upjet-digitalocean/actions/workflows/tag.yaml)
+and click on the `Run workflow` button. Make sure the "Branch" is set to the release branch, e.g., `release-0.2`.
+Enter the version number in the "Release version" field, e.g., `v0.2.1`, and a sensitive value for the "Tag message"
+field. Verify the information and click the "Run workflow" button.
+1. Wait for the Tag Workflow Action to complete.
+1. Run the [CI Workflow Action](https://github.com/crossplane-contrib/provider-upjet-digitalocean/actions/workflows/ci.yml)
+   by clicking on the "Run workflow" button. Make sure the "Branch" is set to the release branch, e.g., `release-0.2`.
+1. Create a new GitHub Release
+   at [New Release](https://github.com/crossplane-contrib/provider-upjet-digitalocean/releases/new).
+   Select the tag that was created by the Tag Workflow Action, e.g., `v0.0.1`, and enter the release notes. Click the
+   "Publish release" button.
+
 ## Explanations
 
+### Release Branches
+
+Release branches are prefixed with `release-` followed by the version number, e.g., `release-0.2`. The version number
+should be composed by the major and minor version numbers, e.g., `0.2`. The patch version number should never be part
+of the release branch name. Patching the existing release branch should be done for patch releases.
+
+#### Creating a Release Branch
+
+Create or patch the [Release Branch](#release-branches) from the `main` branch. You could use visit [Branches](https://github.com/crossplane-contrib/provider-upjet-digitalocean/branches)
+and click on the `New branch` button, or use the following commands:
+
+```shell
+git checkout -b release-<MAJOR>.<MINOR>
+git push origin release-<MAJOR>.<MINOR>
+```
+
+#### Patching a Release Branch
+
+Backport the changes from the `main` branch to the [Release Branch](#release-branches) and push the changes.
```

Comment:
this should be automated by using the Backport github action. e.g. you just apply a label (e.g. `Backport release-0.2`) on a PR to main that you'd like to also backport to a release branch. When the PR is merged to main, the Backport action will automatically run and open up a backport PR that takes the changes and applies them to the release branch.  Should be nice and easy that way letting the tooling do the work for you :)

Reply:
Oooooooooooh I didn't notice this ...... I am keep noticing that I was missing words meaning in this given context because I was not aware of what they meant ... ambiguous language huh!